content/en/blog/2023/otel-dotnet-performance-impacts.md

```diff
@@ -0,0 +1,287 @@
+---
+title: Does OpenTelemetry in .NET Cause Performance Degradation?
+linkTitle: opentelemetry dotnet performance impacts
+date: 2023-03-10
+author: '[MartinDotNet](https://github.com/martinjt) (Honeycomb)'
+---
+
+Contrary to Betteridge’s Law of Tabloid Headlines, the answer is “yes,” but
+context is important. I get this question so often that I thought it was time to
+get some stats on it.
+
+I’ve heard comments like:
+
+- “Instrumenting my code will make it slower”
+- “Activity Object use memory”
+- “If I instrument my code, it will cost me more as I’ll need bigger servers”
+- “OpenTelemetry is immature and isn’t memory efficient”
+- “OpenTelemetry causes performance issues”
+
+I can only assume that these are based on previous versions, or things like
+OpenTracing/OpenCensus (the heritage frameworks that were the feeders for
```

Comment:
I'd avoid mentioning OTracing/Census unless there were known perf issues with them. If yes, please add links to github posts/etc discussing that.

Reply:
Sorry, that comment was for the scene setting part below. This is a widely accepted concept, OpenTracing had memory leaks. I don't believe adding references to that is useful. Acknowledging that there this was an issue, and potentially why that doesn't apply now is definitely relevant.

Reply:
I don't think its a good idea to state OpenTracing/Census has perf issues without solid links backing it.